lib/parser/provenance.cc

```diff
@@ -69,6 +69,18 @@ ProvenanceRange OffsetToProvenanceMappings::Map(std::size_t at) const {
   return provenanceMap_[low].range.Suffix(offset);
 }
 
+std::optional<std::size_t> OffsetToProvenanceMappings::ReverseMap(
+    Provenance at) const {
+  for (std::size_t i = 0; i < provenanceMap_.size(); ++i) {
```

Comment:
You can use a ranged-based for loop here:
`for (const ContiguousProvenanceMapping &mapping : provenanceMap_) {`

In fact, you can simplify it further with structured bindings:
`for (const auto &[start, range] : provenanceMap_) {`